build-tools/automation/azure-pipelines.yaml

```diff
@@ -796,31 +797,95 @@ stages:
 
     - template: yaml-templates/fail-on-issue.yaml
 
-  # Xamarin.Android (Test MSBuild - macOS)
+  # Xamarin.Android (Test MSBuild - macOS - Legacy)
   - template: yaml-templates\run-msbuild-mac-tests.yaml
     parameters:
       node_id: 1
+      job_name: mac_msbuild_tests_1
+      job_suffix: Legacy
 
   - template: yaml-templates\run-msbuild-mac-tests.yaml
     parameters:
       node_id: 2
+      job_name: mac_msbuild_tests_2
+      job_suffix: Legacy
+      run_extra_tests: true
 
   - template: yaml-templates\run-msbuild-mac-tests.yaml
     parameters:
       node_id: 3
+      job_name: mac_msbuild_tests_3
+      job_suffix: Legacy
 
-  # Xamarin.Android (Test MSBuild - Windows)
+  # Xamarin.Android (Test MSBuild - macOS - One .NET)
+  - template: yaml-templates\run-msbuild-mac-tests.yaml
+    parameters:
+      node_id: 1
+      job_name: mac_dotnet_tests_1
+      job_suffix: One .NET
+      nunit_categories: $(DotNetNUnitCategories)
+      nunit_extra: --testparam dotnet=true
+
+  - template: yaml-templates\run-msbuild-mac-tests.yaml
+    parameters:
+      node_id: 2
+      job_name: mac_dotnet_tests_2
+      job_suffix: One .NET
+      nunit_categories: $(DotNetNUnitCategories)
+      nunit_extra: --testparam dotnet=true
+
+  - template: yaml-templates\run-msbuild-mac-tests.yaml
+    parameters:
+      node_id: 3
+      job_name: mac_dotnet_tests_3
+      job_suffix: One .NET
+      nunit_categories: $(DotNetNUnitCategories)
+      nunit_extra: --testparam dotnet=true
+
+  # Xamarin.Android (Test MSBuild - Windows - Legacy)
+  - template: yaml-templates\run-msbuild-win-tests.yaml
+    parameters:
+      node_id: 1
+      job_name: win_msbuild_tests_1
+      job_suffix: Legacy
+
+  - template: yaml-templates\run-msbuild-win-tests.yaml
+    parameters:
+      node_id: 2
+      job_name: win_msbuild_tests_2
+      job_suffix: Legacy
+      run_extra_tests: true
+
+  - template: yaml-templates\run-msbuild-win-tests.yaml
+    parameters:
+      node_id: 3
+      job_name: win_msbuild_tests_3
+      job_suffix: Legacy
+
+  # Xamarin.Android (Test MSBuild - Windows - One .NET)
```

Comment:
I'm starting to worry about fanning out to too many machines on each PR build, especially in the case of Windows where we only have 9 instances available. I think we'll need to address execution scope for `Xamarin.Android.Build.Tests` on PR builds in the somewhat near future. I could go either way here, but we may want to consider only running these new One .NET msbuild test jobs against main/release branches for now as part of this PR?

Reply:
I'm experimenting with adding `condition: eq(variables['RunAllTests'], true)` to all the One .NET tests.

I'll need to queue a build manually and set `RunAllTests` at queue time to verify that works.

Reply:
I think the `condition` is correct now. Here is the manual build: https://build.azdo.io/3948573